solr/core/src/java/org/apache/solr/cli/HealthcheckTool.java

```diff
@@ -227,6 +250,20 @@ protected void runCloudTool(CloudSolrClient cloudSolrClient, CommandLine cli) th
     new JSONWriter(arr, 2).write(report);
     echo(arr.toString());
   }
+
+  @Override
+  public int callTool() throws Exception {
+    if (zkHost == null) {
```

Comment:
i wonder if we should have a more generic abstraction that isn't tied to "zkHost" for detecting if we are in solr cloud mode?   Maybe an explicit method, because I feel like we are slowly hiding zkHost more and more, and the client may just be a solrClient configured with a solrUrl.  <-- this isn't specific to your picocli change I know!